### DISCUSSION TITLE
Can we use app directory for production ?

### DISCUSSION THREAD

**Discussion by Enigma10** [vercel/next.js#46527] [Help]
Hi There, We are migrating our CRA project to next.js. Is the app directory stable enough to use in production?  Since the app directory is in beta. How big new changes will migrate from the beta to the stable version?  
we don't want to migrate to  next12  first and than migrate to next13. 

thanks

**Comment by icyJoseph** [comment]
You can migrate to Next@13 without using appDir though... do not confuse things.

The `appDir` is an experimental feature available in `next@13`, but one can use `next@13` with the pages directory.

I think the pages directory is a great place to start the migration while the appDir becomes stable. Many concepts are still kind of similar, and you'd need to solve assumptions of CSR applications that do not carry over to SSR, anyway.

I think Next.js wants to have appDir ASAP, but you can't/shouldn't take it to prod just yet ~

Instead figure out how your application should behave within an SSR environment, and migrate to pages directory. It won't be fun at all, to migrate a CSR, such as CRA, to SSR and at the same time learn appDir, and Server Components. 

Most of the code from your CRA, turned to pages directory, should be reusable within appDir anyway.

**Reply by eybel** [reply]
> You can migrate to Next@13 without using appDir though... do not confuse things.
> 
> The `appDir` is an experimental feature available in `next@13`, but one can use `next@13` with the pages directory.
> 
> I think the pages directory is a great place to start the migration while the appDir becomes stable. Many concepts are still kind of similar, and you'd need to solve assumptions of CSR applications that do not carry over to SSR, anyway.
> 
> I think Next.js wants to have appDir ASAP, but you can't/shouldn't take it to prod just yet ~
> 
> Instead figure out how your application should behave within an SSR environment, and migrate to pages directory. It won't be fun at all, to migrate a CSR, such as CRA, to SSR and at the same time learn appDir, and Server Components.
> 
> Most of the code from your CRA, turned to pages directory, should be reusable within appDir anyway.

Hi iciJosehp, my question is can we use the new metadata feature (that replaces the head) for my schemas (SEO purpose) in the pages directory in Next.js 13 ? Or I have to use it in the beta "app" directory?

I am not sure if the next13.js features work in the pages directory

thank you

**Comment by Enigma10** [comment]
@icyJoseph  we are already using the pages directory in other projects. It's just that we don't want to migrate two times.  While others might seem to think that migration from `pagesDir` to `appDir` is easy. But It is not easy.

**Reply by icyJoseph** [reply]
Ah that's another story... the thing is that it is not super straightforward because it is still unstable 😿 , I do believe when it hits production mode, it'll be better, not trivial, but it'll just make more sense. 

For example I am overlooking Router events, which are now gone. Or how to do mutations is still a question mark. Layouts might need to be re-designed, etc, but the important thing is that you'd seldom run into "implementation issues" proper of a beta stage, and rather you'd be able to focus on migrating your app.
